### PR TITLE
[IMP] website: keyword search base on language on website promote

### PR DESCRIPTION
**Pull request by gan-odoo**
Task : https://www.odoo.com/web#id=8805&view_type=form&model=project.task&menu_id=3942&action=327
Pad : https://pad.odoo.com/p/r.QSI79TcaBa81Jnmc

Note :  Add language selection box below keyword box on website promote. 
Language box contain register language on website and keyword are search using this language.

### CODE REVIEW

**Comment by Gorash** [comment]
:+1: 
@tde-banana-odoo you can merge

File: addons/website/models/ir_ui_view.py [hunk 1]
```diff
@@ -13,12 +13,10 @@
 
 
 class view(osv.osv):
-    _inherit = "ir.ui.view"
+    _name = "ir.ui.view"
+    _inherit = ["ir.ui.view", "website.seo.metadata"]
```
Comment (tde-banana-odoo) [inline addons/website/models/ir_ui_view.py]: This change should be in a separate commit. Explain why you did this (not what you did, but why you did it).

File: addons/website/static/src/js/website.seo.js [hunk 2]
```diff
@@ -71,31 +77,39 @@ var SuggestionList = Widget.extend({
...
+        self.$el.append(_t("Loading..."));
+        var language = self.language || website.get_context().lang.toLowerCase();
+        ajax.jsonRpc('/website/seo_suggest', 'call', {
+            'keywords': self.root,
+            'lang': language,
+        }).then(function(list){
+            self.addSuggestions(JSON.parse(list));
+        });
     },
+    addSuggestions: function(list) {
```
Comment (tde-banana-odoo) [inline addons/website/static/src/js/website.seo.js]: list is not a viable variable name.

File: addons/website/static/src/js/website.seo.js [hunk 3]
```diff
@@ -71,31 +77,39 @@ var SuggestionList = Widget.extend({
...
+        var self = this;
+        self.$el.empty();
+        // TODO Improve algorithm + Ajust based on custom user keywords
+        var regex = new RegExp(self.root, "gi");
+        var cleanList = _.map(list, function (word) {
+            return word.replace(regex, "").trim();
+        });
+        // TODO Order properly ?
+        _.each(_.uniq(cleanList), function (keyword) {
```
Comment (tde-banana-odoo) [inline addons/website/static/src/js/website.seo.js]: Ensure that cleanList is already cleaned. You should not have to clean (unify) something already cleaned.

Comment (tde-banana-odoo) [inline addons/website/static/src/js/website.seo.js]: Moreover remove if (keyword) and ensure the clean list effectively has not-void entries.

File: addons/website/static/src/xml/website.seo.xml [hunk 4]
```diff
@@ -4,6 +4,13 @@
     <t t-name="website.editorbar.menu.promote">
         <li><a data-action="promote-current-page" href="#"><span title="Promote page on the web">Promote</span></a></li>
     </t>
+
+    <t t-name="Configurator.language_promot">
```
Comment (tde-banana-odoo) [inline addons/website/static/src/xml/website.seo.xml]: language_promot ?

File: addons/website/static/src/xml/website.seo.xml [hunk 5]
```diff
@@ -4,6 +4,13 @@
...
+        <t t-foreach="language" t-as="lang">
+            <option t-att-value="lang[0]" selected="selected" t-if="lang[0] == def_lang"><t t-esc="lang[1]" /></option>
+            <option t-att-value="lang[0]" t-if="lang[0] != def_lang"><t t-esc="lang[1]" /></option>
```
Comment (Gorash) [inline addons/website/static/src/xml/website.seo.xml]: t-att-selected="lang[0] == def_lang ? 'selected' : null"
doesn't work ?

Comment (gan-odoo) [inline addons/website/static/src/xml/website.seo.xml]: I agree with you.
Done

File: addons/website/static/src/js/website.seo.js (hunk 2 above)
Comment (gan-odoo) [inline addons/website/static/src/js/website.seo.js]: Done

File: addons/website/models/ir_ui_view.py (hunk 1 above)
Comment (gan-odoo) [inline addons/website/models/ir_ui_view.py]: Metadata related field are also available in website.seo.metadata model so remove fields from ir.ui.view and inherit website.seo.metadata.
OK. i have put in separate commit

File: addons/website/static/src/js/website.seo.js (hunk 3 above)
Comment (gan-odoo) [inline addons/website/static/src/js/website.seo.js]: cleanList is not clean, keyword list contain with root keyword, so remove this root keyword from every suggested keyword.
and replace a root keyword to null string so need to check condition for keyword otherwise show 1 keyword tag without string.

File: addons/website/static/src/xml/website.seo.xml (hunk 4 above)
Comment (gan-odoo) [inline addons/website/static/src/xml/website.seo.xml]: It is template of selection box that contain a languages which are define on website.

Comment (tde-banana-odoo) [inline addons/website/static/src/xml/website.seo.xml]: I agree. But don't you spot any typo here ?

File: addons/website/static/src/js/website.seo.js (hunk 3 above)
Comment (tde-banana-odoo) [inline addons/website/static/src/js/website.seo.js]: Then when creating clean_list (no camel case), remove void elements. You have to correctly prepare the data you want to use before actually use them.

File: addons/website/static/src/xml/website.seo.xml (hunk 4 above)
Comment (gan-odoo) [inline addons/website/static/src/xml/website.seo.xml]: OK Done!

File: addons/website/static/src/js/website.seo.js (hunk 3 above)
Comment (gan-odoo) [inline addons/website/static/src/js/website.seo.js]: Done

**Comment by tde-banana-odoo** [comment]
Merged at https://github.com/odoo/odoo/commit/d52ec7fc42cbeecafea921ea30d1d0723a59078a and https://github.com/odoo/odoo/commit/0cbf2e596f2d7ff2fc630fb70796373cab91e488 .
